src/config/__tests__/settings.spec.ts

```diff
@@ -0,0 +1,228 @@
+import fs from 'fs';
+
+import * as settings from '../settings';
+import { PathManagementStrategy } from '@/integrations/pathManager';
+
+describe('updateFromCommandLine', () => {
+  let prefs: settings.Settings;
+  let origPrefs: settings.Settings;
+
+  beforeEach(() => {
+    jest.spyOn(fs, 'writeFileSync').mockImplementation(() => { });
+    prefs = {
+      version:    4,
+      kubernetes: {
+        version:                    '1.23.5',
+        memoryInGB:                 4,
+        numberCPUs:                 2,
+        port:                       6443,
+        containerEngine:            settings.ContainerEngine.MOBY,
+        checkForExistingKimBuilder: false,
+        enabled:                    true,
+        WSLIntegrations:            {},
+        options:                    {
+          traefik: true,
+          flannel: false,
+        },
+        suppressSudo: false
+      },
+      portForwarding: { includeKubernetesServices: false },
+      images:         {
+        showAll:   true,
+        namespace: 'k8s.io'
+      },
+      telemetry:              true,
+      updater:                true,
+      debug:                  true,
+      pathManagementStrategy: PathManagementStrategy.NotSet,
+    };
+    origPrefs = JSON.parse(JSON.stringify(prefs));
+  });
+
+  describe('getUpdatableNode', () => {
+    test('returns null on an invalid top level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'blah-blah-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns null on an invalid internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-blah');
+
+      expect(result).toBeNull();
+    });
+    test('returns the full pref with a top-level accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes') as [Record<string, any>, string];
+
+      expect(result).not.toBeNull();
+      const [lhs, accessor] = result;
+
+      expect(lhs).toEqual(prefs);
+      expect(accessor).toBe('kubernetes');
+    });
+    test('returns a partial pref with an internal accessor', () => {
+      const result = settings.getUpdatableNode(prefs, 'kubernetes-options-flannel') as [Record<string, any>, string];
+
+      expect(result).not.toBeNull();
+      const [lhs, accessor] = result;
+      const flannelNow = prefs.kubernetes.options.flannel;
+      const flannelAfter = !flannelNow;
+
+      expect(lhs).toEqual({
+        traefik: prefs.kubernetes.options.traefik,
+        flannel: flannelNow,
+      });
```

Comment:
This would be more future proof (in case we get more settings here) as:
```suggestion
      expect(lhs).toEqual({
        ...prefs.kuberntes.options,
        flannel: flannelNow,
      });
```
(Of course you _can_ also just check `.toEqual(prefs.kubernetes.options`)

Reply:
Should be testing `...origPrefs.kubernetes.options`